app/src/main/java/org/jellyfin/mobile/player/PlayerViewModel.kt

```diff
@@ -168,6 +182,30 @@ class PlayerViewModel(application: Application) : AndroidViewModel(application),
                 else -> DefaultRenderersFactory.EXTENSION_RENDERER_MODE_ON
             }
             setExtensionRendererMode(rendererMode)
+            // set the media selector
+            setMediaCodecSelector { mimeType, requiresSecureDecoder, requiresTunnelingDecoder ->
+                val decoderInfoList = MediaCodecSelector.DEFAULT.getDecoderInfos(mimeType, requiresSecureDecoder, requiresTunnelingDecoder)
+                // allow decoder selection only for video
+                if (!MimeTypes.isVideo(mimeType)) {
+                    return@setMediaCodecSelector decoderInfoList
+                }
```

Comment:
We don't need to check this, since this Player is only used for video anyway.
```suggestion
```

Reply:
Yeah, but the thing is the video will have audio, and subtitle tracks as well. So this function will be called for each of them and I added that to ensure we only change the decoder of the video track.

Reply:
Oh, that makes sense, then I misunderstood this. Thanks for the explanation!